internal/db/update/schema.go

```diff
@@ -126,25 +140,31 @@ func (s *SchemaUpdate) Ensure(db *sql.DB) (int, error) {
 func (s *SchemaUpdate) Dump(db *sql.DB) (string, error) {
 	var statements []string
 	err := query.Transaction(context.TODO(), db, func(ctx context.Context, tx *sql.Tx) error {
-		versions, err := query.SelectIntegers(ctx, tx, "SELECT version FROM schemas ORDER BY version")
-		if err != nil {
-			return err
-		}
+		currentVersion := 0
+		for versionType := range s.updates {
```

Comment:
Do we need to interate internal updates first before external ones? Like this the order is random. Does it make a difference?

Reply:
Thanks, I overlooked that. I think for sanity's sake the internal updates should come first, but either case has trade-offs.

- If all internal updates come first, but an external update depends on an internal table, a new internal update can it can change the table from under that pre-existing external update unexpectedly.

- If internal updates don't come first, then an external update can change an internal table and then the following internal updates could break.

It speaks to the larger problem that we have no way of restricting external schema updates to not depend on internal ones.

Reply:
Did this get addressed?

Reply:
Yes, here: https://github.com/canonical/microcluster/pull/94/files#diff-268ba13a60418730fa859d08731781b80293f421148bcfc6e507f8f2f8a95709R139-R141